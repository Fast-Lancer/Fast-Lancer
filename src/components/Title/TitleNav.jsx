import styles from './Title.module.css'
import { Link } from 'react-router-dom'

export default function TitleNav({ pageTitle, pageHeader }) {
  switch (pageTitle) {
    case 'projects':
      return (
        <ul className={styles.tabnav}>
          <li className={styles.active}>
            <a href={' '}>{pageHeader}</a>
          </li>
          <li>
            <Link to="/clients">Clients</Link>
          </li>
          <li>
            <Link to="/projects/new">New Project</Link>
          </li>
        </ul>
      )

    case 'clients':
      return (
        <ul className={styles.tabnav}>
          <li className={styles.active}>
            <a href={' '}>{pageTitle}</a>
          </li>
          <li>
            <Link to="/projects">Projects</Link>
          </li>
          <li>
            <Link to="/clients/newclient">New Client</Link>
          </li>
        </ul>
      )

    case 'project detail':
      return (
        <ul className={styles.tabnav}>
          <li className={styles.active}>
            <a href={'/projects'}>Projects</a>
          </li>
          <li>
            <Link to="/clients">Clients</Link>
          </li>
          <li>
            <Link to="/editproject">Edit Project</Link>
          </li>
        </ul>
      )

    case 'client detail':
      return (
        <ul className={styles.tabnav}>
          <li className={styles.active}>
            <a href={'/clients'}>Clients</a>
          </li>
          <li>
            <Link to="/projects">Projects</Link>
          </li>
          <li>
            <Link to="/editclient">Edit Client</Link>
          </li>
        </ul>
      )

    case 'new-edit project':
      return (
        <ul className={styles.tabnav}>
          <li className={styles.active}>
            <a href={'/projects'}>Projects</a>
          </li>
          <li>
            <Link to="/clients">Clients</Link>
          </li>
        </ul>
      )

    case 'new-edit client':
      return (
        <ul className={styles.tabnav}>
          <li className={styles.active}>
            <a href={'/clients'}>Clients</a>
          </li>
          <li>
            <Link to="/projects">Projects</Link>
          </li>
        </ul>
      )

    default:
      return 'err'
  }
}
