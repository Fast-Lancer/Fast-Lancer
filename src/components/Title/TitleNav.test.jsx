import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import TitleNav from './TitleNav'

it('renders the TitleNav component accurately', () => {
  const { container } = render(
    <Router>
      <TitleNav pageTitle='projects' pageHeader='Header' />
    </Router>
  )
  expect(container).toMatchSnapshot()
})
