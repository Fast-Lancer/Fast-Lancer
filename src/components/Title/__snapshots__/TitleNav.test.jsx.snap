// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TitleNav component accurately 1`] = `
<div>
  <ul
    class="tabnav"
  >
    <li
      class="active"
    >
      <a
        href=" "
      >
        Header
      </a>
    </li>
    <li>
      <a
        href="/clients"
      >
        Clients
      </a>
    </li>
    <li>
      <a
        href="/projects/new"
      >
        New Project
      </a>
    </li>
    <li>
      <a
        href="/login"
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;
